sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -0,0 +1,187 @@
+        """
+        async with self._lock:
+            error = None
+            if (consumer_group, '-1') in self._event_processors:
+                error = ValueError("This consumer client is already receiving events from all partitions for"
+                                 " consumer group {}. "
+                                 "Shouldn't receive from any other partitions again".format(consumer_group))
+            elif partition_id is None and any(x[0] == consumer_group for x in self._event_processors):
+                error = ValueError("This consumer client is already receiving events for consumer group {}. "
+                                 "Shouldn't receive from all partitions again".format(consumer_group))
+            elif (consumer_group, partition_id) in self._event_processors:
+                error = ValueError("This consumer is already receiving events from partition {} for consumer group {}. "
+                                 "Shouldn't receive from it again.".format(partition_id, consumer_group))
+            if error:
+                log.warning(error)
+                raise error
+
+            event_processor = EventProcessor(
+                self, consumer_group, on_event,
+                partition_id=partition_id,
+                partition_manager=self._partition_manager,
+                error_handler=on_error,
+                partition_initialize_handler=on_partition_initialize,
+                partition_close_handler=on_partition_close,
+                initial_event_position=initial_event_position or EventPosition("-1"),
+                polling_interval=self._load_balancing_interval,
+                owner_level=owner_level,
+                prefetch=prefetch,
+                track_last_enqueued_event_properties=track_last_enqueued_event_properties,
+            )
+            if partition_id:
+                self._event_processors[(consumer_group, partition_id)] = event_processor
+            else:
+                self._event_processors[(consumer_group, "-1")] = event_processor
+        try:
+            await event_processor.start()
+        finally:
+            await event_processor.stop()
+            async with self._lock:
+                if partition_id and (consumer_group, partition_id) in self._event_processors:
+                    del self._event_processors[(consumer_group, partition_id)]
+                elif partition_id is None and (consumer_group, '-1') in self._event_processors:
+                    del self._event_processors[(consumer_group, "-1")]
+
+    async def close(self):
+        # type: () -> None
+        """Stop retrieving events from event hubs and close the underlying AMQP connection and links.
+
+        """
+        async with self._lock:
+            for _ in range(len(self._event_processors)):
+                _, ep = self._event_processors.popitem()
+                await ep.stop()
+            await super().close()
```

Comment:
I think super should be called with args: `super(EventHubConsumerClient, self).close()`

Reply:
Super() works in Python3. But I'll change to have args.

Reply:
That's acceptable syntax in Python 3 only code ;)